apps/ejabberd/src/event_pusher/mod_event_pusher_hook_translator.erl

```diff
@@ -0,0 +1,106 @@
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(mod_event_pusher_hook_translator).
+-author('konrad.zemek@erlang-solutions.com').
+
+-behaviour(gen_mod).
+
+-include("jlib.hrl").
+-include("mod_event_pusher_events.hrl").
+
+-export([start/2, stop/1]).
+-export([user_send_packet/4, filter_packet/1, user_present/2, user_not_present/5]).
+
+%%--------------------------------------------------------------------
+%% gen_mod API
+%%--------------------------------------------------------------------
+
+-spec start(Host :: ejabberd:server(), Opts :: proplists:proplist()) -> any().
+start(Host, _Opts) ->
+    ejabberd_hooks:add(rest_user_send_packet, Host, ?MODULE, user_send_packet, 90),
```

Comment:
I suggest using `ejabberd_hooks:add/1` and `ejabberd_hooks:delete/1`. They accept list of hooks. Example is in [mod_roster](https://github.com/esl/MongooseIM/blob/master/apps/ejabberd/src/mod_roster.erl#L174)